packages/data/src/redux-store/test/combine-reducers.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal dependencies
+ */
+import { combineReducers } from '../combine-reducers';
+
+function counterReducer( count = 0, action ) {
+	if ( action.type === 'INC' ) {
+		return count + 1;
+	}
+	return count;
+}
+
+describe( 'combineReducers', () => {
+	it( 'initializes state', () => {
+		const reducer = combineReducers( { foo: counterReducer } );
+		expect( reducer( undefined, { type: 'INIT' } ) ).toEqual( { foo: 0 } );
+	} );
+
+	it( 'dispatches actions to subreducers', () => {
+		const reducer = combineReducers( { foo: counterReducer } );
+		expect( reducer( { foo: 1 }, { type: 'INC' } ) ).toEqual( { foo: 2 } );
+	} );
+
+	it( 'returns identical state when there is no change', () => {
+		const reducer = combineReducers( { foo: counterReducer } );
+		const state = { foo: 1 };
+		expect( reducer( state, { type: 'NOOP' } ) ).toBe( state );
+	} );
+
+	it( 'returns identical substate when there is no change', () => {
+		const reducer = combineReducers( {
+			foo: counterReducer,
+			bar: combineReducers( { baz: ( s ) => s } ),
+		} );
+		const prevState = { foo: 1, bar: { baz: 1 } };
+		const nextState = reducer( prevState, { type: 'INC' } );
+		expect( nextState.foo ).toBe( 2 ); // changed
+		expect( nextState.bar ).toBe( prevState.bar ); // not changed
+	} );
+
+	it( 'does not mind undefined state', () => {
+		const reducer = combineReducers( { foo: ( s ) => s } );
+		// combineReducers from Redux would throw an exception when a reducer returns
+		// `undefined` state, but we don't mind and treat is as any other value.
+		expect( reducer( undefined, { type: 'INIT' } ) ).toEqual( {
+			foo: undefined,
+		} );
+	} );
+
+	it( 'does not mind unknown state keys', () => {
```

Comment:
Redux's `combineReducers` will also complain if there are no reducer subkeys. We could add a test for that too, although it would be a rare scenario.

Reply:
I don't understand, do you mean calling `combineReducers( {} )`? I think that's a valid case, it returns a valid "zero" reducer that always returns `{}`. Something like empty `{}` TypeScript type.

Reply:
Yes, I was talking about `combineReducers( {} )`. AFAIK the Redux version won't allow this, while our version allows it. So that makes it a good case for a unit test IMHO.